byzcoin/bcadmin/main.go

```diff
@@ -80,10 +84,145 @@ var cmds = cli.Commands{
 		Action: add,
 	},
 	{
-		Name:    "keys",
-		Usage:   "creates a new key pair, which may be used with add",
+		Name:    "key",
+		Usage:   "generates a new keypair and prints the public key in the stdin",
 		Aliases: []string{"k"},
-		Action:  keys,
+		Flags: []cli.Flag{
+			cli.StringFlag{
+				Name:  "save",
+				Usage: "file in which the user wants to save the public key instead of printing it",
+			},
+		},
+		Action: key,
+	},
+	{
+		Name:    "darc_add",
+		Usage:   "add a new darc to the instance",
+		Aliases: []string{"da"},
+		Flags: []cli.Flag{
+			cli.StringFlag{
+				Name:   "bc",
+				EnvVar: "BC",
+				Usage:  "the ByzCoin config to use",
+			},
+			cli.StringFlag{
+				Name:  "owner",
+				Usage: "owner of the darc allowed to sign and evolve it",
+			},
+			cli.StringFlag{
+				Name:  "darc",
+				Usage: "darc from which we create the new darc - genesis if not mentioned",
+			},
+			cli.StringFlag{
+				Name:  "sign",
+				Usage: "signature for the generating darc",
+			},
+			cli.StringFlag{
+				Name:  "out_id",
+				Usage: "output file for the darc id",
+			},
+			cli.StringFlag{
+				Name:  "out_desc",
+				Usage: "output file for the darc description",
+			},
+			cli.StringFlag{
+				Name:  "out_key",
+				Usage: "output file for the darc key",
+			},
+		},
+		Action: darc_add,
+	},
+	{
+		Name:    "darc_show",
+		Usage:   "shows darc with given ID",
+		Aliases: []string{"ds"},
+		Flags: []cli.Flag{
+			cli.StringFlag{
+				Name:   "bc",
+				EnvVar: "BC",
+				Usage:  "the ByzCoin config to use",
+			},
+			cli.StringFlag{
+				Name:  "id",
+				Usage: "the id of the darc one wants to show",
+			},
+			cli.StringFlag{
+				Name:  "save",
+				Usage: "file in which the user wants to save the darc",
+			},
+		},
+		Action: darc_show,
+	},
+	{
+		Name:    "darc_rule_add",
+		Usage:   "add a rule and signer to a darc",
+		Aliases: []string{"dra"},
+		Flags: []cli.Flag{
+			cli.StringFlag{
+				Name:   "bc",
+				EnvVar: "BC",
+				Usage:  "the ByzCoin config to use",
+			},
+			cli.StringFlag{
+				Name:  "identity",
+				Usage: "the identity of the signer who will be allowed to access the contract (e.g. ed25519:a35020c70b8d735...0357))",
+			},
+			cli.StringFlag{
+				Name:  "darc",
+				Usage: "the darc to which we want to add a rule and signer",
+			},
+			cli.StringFlag{
+				Name:  "signer",
+				Usage: "the signer of the operation",
```

Comment:
Make it clearer that this is the filename of the private key file. (I think; is that what this is? If not, where does the private key get set?)